app/_clients/champions-client/champions-client.ts

```diff
@@ -0,0 +1,36 @@
+import ChampionsData from "../../_data/champions";
+
+const _ChampionsClient: ChampionsClient = (() => {
+  const allChampions = ChampionsData.data;
+  const allChampionsArray = Object.values(allChampions);
+
+  function getChampion(championId: string): ChampionData {
+    return ChampionsData.data[championId];
+  }
+
+  function searchChampions(searchTerm: string): ChampionData[] {
+    let searchRegex: RegExp = new RegExp(searchTerm.split("").join(".*"), "i");
+
+    return Object.values(ChampionsData.data).filter(
+      (champion) =>
+        champion.id.match(searchRegex) || champion.name.match(searchTerm),
+    );
+  }
+
+  function getChampions(): { [championId: string]: ChampionData } {
+    return allChampions;
+  }
+
+  function getChampionsArray(): ChampionData[] {
+    return allChampionsArray;
+  }
+
+  return {
+    getChampion,
+    getChampions,
+    getChampionsArray,
+    searchChampions,
+  };
+})();
+
+export default _ChampionsClient;
```

Comment:
I would suggest to try the react hooks path instead of invoked function

```suggestion
const useChampionsClient: ChampionsClient = () => {
  const allChampions = ChampionsData.data;
  const allChampionsArray = Object.values(allChampions);

  function getChampion(championId: string): ChampionData {
    return ChampionsData.data[championId];
  }

  function searchChampions(searchTerm: string): ChampionData[] {
    let searchRegex: RegExp = new RegExp(searchTerm.split("").join(".*"), "i");

    return Object.values(ChampionsData.data).filter(
      (champion) =>
        champion.id.match(searchRegex) || champion.name.match(searchTerm),
    );
  }

  function getChampions(): { [championId: string]: ChampionData } {
    return allChampions;
  }

  function getChampionsArray(): ChampionData[] {
    return allChampionsArray;
  }

  return {
    getChampion,
    getChampions,
    getChampionsArray,
    searchChampions,
  };
};

export default useChampionsClient;
```

doing so would allow you to use this as:
```ts
const { getChampionsArray } = useChampionsClient();
```